src/XAxis.js

```diff
@@ -3,13 +3,34 @@ import PropTypes from "prop-types";
 import React from "react";
 import { getAxisChildProps } from "./utils/Axis";
 import { sumMargins } from "./utils/Margin";
-import { getTickDomain } from "./utils/Scale";
+import { getTickDomain, inferScaleType, invertPointScale } from "./utils/Scale";
 import xyPropsEqual from "./utils/xyPropsEqual";
 import XAxisLabels from "./XAxisLabels";
 import XAxisTitle from "./XAxisTitle";
 import XGrid from "./XGrid";
 import XTicks from "./XTicks";
 
+function getMouseOptions(event, xScale) {
```

Comment:
you're using this twice in two places, go ahead and add this to a new `utils` file